### ISSUE TITLE
Too Many SAM Deploys in Parallel results in Rate Limit

### ISSUE THREAD

**Issue by reskin89** [aws/aws-sam-cli#2991]
### Description:
When deploying multiple lambdas in Parallel via CI, some of the deploys will fail, as the Cloudformation deploy runs the "DescribeStack" api call, and only so many of these calls are allowed in a given time frame, causing a failure of some stacks to fail and some to succeed.

### Steps to reproduce:
Create a SAM deploy/template per lambda, and deploy them all in parallel, it seems to start to arise once you hit 7+ lambdas

### Observed result:
The lambdas that start their deploy "last" will encounter "Error Deploy failed, DescribeStacks Rate Limit Exceeded'

### Expected result:
All Cloudformation Stacks deploy successfully

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: NA
2. `sam --version`: SAM CLI, version 1.23.0
3. AWS region: Any US Region

**Comment by GregTurner** [comment]
Second this, ideally SAM CLI should retry with exponential backoff AND provide the caller of SAM CLI with the ability to set the values for total time to retry.

**Comment by brian-payne2** [comment]
+1, this rate limit issue causes my SAM deploys to fail. An exponential backoff would be extremely helpful as AWS can't expand the read capacity limitations on those calls.

**Comment by jfuss** [comment]
Duplicate of https://github.com/aws/aws-sam-cli/issues/2671 Closing.
